File: js/anthropic/streaming.ts
import Anthropic from "@anthropic-ai/sdk";
import "dotenv/config";
import { traceable } from "langsmith/traceable";

const client = new Anthropic();

const streamingMessage = traceable(async function streamingMessage(
  messages: Anthropic.Messages.MessageParam[]
) {
  console.log("Starting streaming...\n");

  const stream = await client.messages.stream({
    model: "claude-sonnet-4-5",
    max_tokens: 1024,
    messages,
  });

  for await (const event of stream) {
    if (
      event.type === "content_block_delta" &&
      event.delta.type === "text_delta"
    ) {
      process.stdout.write(event.delta.text);
    }
  }

  console.log("\n\nStreaming complete!");

  const finalMessage = await stream.finalMessage();
  return finalMessage;
});

export async function main() {
  return streamingMessage([
    {
      role: "user",
      content: "Write a short poem about artificial intelligence.",
    },
  ]);
}

export { streamingMessage };

File: python/wrap_anthropic/streaming.py
import anthropic
from langsmith import traceable
from langsmith.wrappers import wrap_anthropic

client = wrap_anthropic(anthropic.Anthropic())


@traceable(name="Streaming Message")
def streaming_message():
    """Stream a message response."""
    print("Starting streaming...")

    with client.messages.stream(
        model="claude-sonnet-4-5",
        max_tokens=1024,
        messages=[
            {
                "role": "user",
                "content": "Write a short poem about artificial intelligence.",
            }
        ],
    ) as stream:
        for text in stream.text_stream:
            print(text, end="", flush=True)

    print("\n\nStreaming complete!")
    return stream.get_final_message()


def main():
    print("Running wrap_anthropic streaming example...")
    streaming_message()
    return {"streaming": "complete"}
